modules/stochastic_tools/include/distributions/TruncatedNormalDistribution.h

```diff
@@ -25,9 +25,25 @@ class TruncatedNormalDistribution : public NormalDistribution
 public:
   TruncatedNormalDistribution(const InputParameters & parameters);
 
-  virtual Real pdf(const Real & x) override;
-  virtual Real cdf(const Real & x) override;
-  virtual Real quantile(const Real & p) override;
+  virtual Real pdf(const Real & x) const override;
+  virtual Real cdf(const Real & x) const override;
+  virtual Real quantile(const Real & p) const override;
+
+  Real pdf(const Real & x,
+           const Real & mean,
+           const Real & std_dev,
+           const Real & lower_bound,
+           const Real & upper_bound) const;
```

Comment:
I'm curious, are you actually sharing memory with these references? If not you might consider making these just Real parameters. Sometimes copies are actually faster than references due to locality. No big deal, just a thought.

Reply:
These could be copies.